### ISSUE TITLE
vq objective loss missing?

### ISSUE THREAD

**Issue by fzargarbashi** [heyuanYao-pku/MoConVQ#11]
Hi,

I am trying to reproduce your paper results. As I was going through the code, I couldn't find the implementation/usage of the vq objective loss (third term from eq. 6 in the paper). It seems like only the reconstruction, commitment and regularization losses are used for training. Is that correct? Why is that term not used?

**Comment by heyuanYao-pku** [comment]
It's not missing. In practice, most paper will update the codebook during the forward process. The loss is optimized with moving average instead of the gradient descent. It's a common setting and we discussed in the equ(10) in the paper.
